superset-frontend/src/explore/components/controls/MetricControl/AdhocMetric.js

```diff
@@ -80,9 +80,9 @@ export default class AdhocMetric {
 
     this.optionName =
       adhocMetric.optionName ||
-      `metric_${Math.random()
+      `metric_${Math.random().toString(36).substring(2, 15)}_${Math.random()
```

Comment:
This logic is also used [here](https://github.com/apache/superset/pull/17427/files#diff-c3790e716d19041bf895c7134e1961907442bd0d8477d6e897f33517591832e9R124) - maybe we could abstract it to a shared utility